virtcontainers/hypervisor.go

```diff
@@ -552,9 +552,9 @@ type hypervisor interface {
 	pauseSandbox() error
 	saveSandbox() error
 	resumeSandbox() error
-	addDevice(devInfo interface{}, devType types.DeviceType) error
-	hotplugAddDevice(devInfo interface{}, devType types.DeviceType) (interface{}, error)
-	hotplugRemoveDevice(devInfo interface{}, devType types.DeviceType) (interface{}, error)
+	addDevice(types.Device) error
```

Comment:
Can you elaborate how this change will help decouple the hypervisor from the endpoint dependency?
From what I understand here, we're simply moving the device info as part of the device structure itself, but the detection of the device types such as `Endpoint` is still happening inside each hypervisor implementation, right?